ui/plugins/eu.esdihumboldt.hale.ui/src/eu/esdihumboldt/hale/ui/service/entity/OptionalPropertiesFilter.java

```diff
@@ -50,11 +62,39 @@ public boolean select(Viewer viewer, Object parentElement, Object element) {
 			if (def instanceof PropertyDefinition) {
 				Cardinality cardinality = ((PropertyDefinition) def)
 						.getConstraint(Cardinality.class);
-				return !(cardinality.getMinOccurs() == 0 && ((PropertyDefinition) def)
-						.getConstraint(NillableFlag.class).isEnabled());
+				if (cardinality.getMinOccurs() == 0)
+					return false;
+
+				if (((PropertyDefinition) def).getConstraint(NillableFlag.class).isEnabled()) {
+					return !areChildrenOptional(entityDef);
+				}
 			}
 		}
 		return true;
 	}
 
+	private boolean areChildrenOptional(EntityDefinition entityDef) {
+
+		Collection<? extends EntityDefinition> children = eds.getChildren(entityDef);
```

Comment:
Using the `EntityDefinitionService` here is a bit overkill, as it will also retrieve children with contexts, which don't have to be checked (as they are just duplicates of other children with the same info). But it shouldn't be to much additional effort as usually there are not many contexts - so you can leave it like it is if you like, but it would be good if you add a comment there stating that definitions are checked that don't need to and that this might be improved.